pydicom/pixel_data_handlers/rle_handler.py

```diff
@@ -203,17 +203,18 @@ def _rle_decode_plane(data):
 
     while pos < len_data:
         header_byte = data[pos]
-
         pos += 1
-
         if header_byte > 128:
-            result.extend([data[pos]] * (255 - header_byte + 2))
+            # Extend by copying the next byte (-N + 1) times
+            # however since using uint8 instead of int8 this will be
+            # (256 - N + 1) times
+            result.extend(data[pos:pos + 1] * (257 - header_byte))
             pos += 1
```

Comment:
With python 2.7
```python
>>> a = b'\x01\x02\x03'
>>> [a[1]] * 3
['\x02', '\x02', '\x02']
>>> a[1:2] * 3
'\x02\x02\x02'
```
Removes the overhead from creating a list every time that condition runs